extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/base/ModeledDirectoryObjectService.java

```diff
@@ -387,25 +396,36 @@ public Collection<InternalType> retrieveObjects(ModeledAuthenticatedUser user,
             Collection<String> identifiers) throws GuacamoleException {
 
         // Ignore invalid identifiers
-        identifiers = filterIdentifiers(identifiers);
+        List<String> filteredIdentifiers = filterIdentifiers(identifiers);
 
         // Do not query if no identifiers given
-        if (identifiers.isEmpty())
+        if (filteredIdentifiers.isEmpty())
             return Collections.<InternalType>emptyList();
 
-        Collection<ModelType> objects;
+        int batchSize = environment.getDefaultBatchSize();
 
-        // Bypass permission checks if the user is privileged
-        if (user.isPrivileged())
-            objects = getObjectMapper().select(identifiers);
+        Collection<ModelType> allObjects = new ArrayList<>(filteredIdentifiers.size());
+
+        for (int startIndex = 0; startIndex < filteredIdentifiers.size(); startIndex += batchSize) {
```

Comment:
A comment before this loop would be very handy for reader's to understand what's going on here. There's a lot of dense code that's understandable with some thinking, but at a glance it's not super obvious (at least not to me).

Reply:
That said, there's also probably easier ways to do this (splitting a list into chunks). Have a look at https://guava.dev/releases/19.0/api/docs/com/google/common/collect/Lists.html#partition(java.util.List,%20int) for example (the JDBC extensions already include guava as a dependency).

Reply:
I switched to a more functional map reduce style of code

Reply:
I do dig functional-style programming, but this still looks a little overly complex. Did you look into the function I mentioned above? All this IntStream + filter + map stuff looks like it could be replaced by

```
        Collection<ModelType> allObjects = Lists.partition(filteredIdentifiers, batchSize).stream()
                .flatMap(chunk -> {
                     ...
```